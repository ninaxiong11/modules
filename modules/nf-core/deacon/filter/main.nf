process DEACON_FILTER {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/deacon:0.5.0--h4349ce8_0':
        'biocontainers/deacon:0.5.0--h4349ce8_0' }"

    input:
    tuple val(meta), path(index), path(reads)

    output:
    tuple val(meta), path("${prefix}*.fq") , emit: fastq_filtered
    tuple val(meta), path("${prefix}.json"), emit: log
    path "versions.yml"			   , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    def read_type = (reads instanceof List) ? "-o ${prefix}_1.fq -O ${prefix}_2.fq" : "> ${prefix}.fq"

    """
    deacon \\
        filter \\
        --threads ${task.cpus} \\
        $args \\
	    --summary ${prefix}.json \\
        -d $index \\
        $reads \\
        ${read_type}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        deacon: \$(deacon --version | head -n1 | sed 's/deacon //g')
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.fq
    touch ${prefix}.json

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        deacon: \$(deacon --version | head -n1 | sed 's/deacon //g')
    END_VERSIONS
    """
}
