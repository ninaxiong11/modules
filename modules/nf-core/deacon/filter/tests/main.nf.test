nextflow_process {

    name "Test Process DEACON_FILTER"
    script "../main.nf"
    process "DEACON_FILTER"

    tag "modules"
    tag "modules_nfcore"
    tag "deacon"
    tag "deacon/filter"
    tag "deacon/index"

    test("sarscov2 - fastq - single-end") {

        setup {
            run("DEACON_INDEX") {
                script "../../index/main.nf"
                process {
                    """
                    input[0] = [
                                [ id:'test', single_end:true ], // meta map
                                file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                    ]
                    """
                }
            }
        }

        when {
            process {
                """
                filter_fastq_ch = Channel.of(
                    [
                        [ id:'test', single_end:true ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true)
                    ]
                )
                input[0] = DEACON_INDEX.out.index.join(filter_fastq_ch)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.log[0][1]).name,
                    process.out.fastq_filtered,
                    process.out.versions.collect { path(it).yaml }
                ).match()}
            )
        }

    }

    test("sarscov2 - fastq - paired-end") {

        setup {
            run("DEACON_INDEX") {
                script "../../index/main.nf"
                process {
                    """
                    input[0] = [
                                [ id:'test', single_end:false ], // meta map
                                file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                    ]
                    """
                }
            }
        }

        when {
            process {
                """
                filter_fastq_ch = Channel.of(
                    [
                        [ id:'test', single_end:false ], // meta map
                        [
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                        ]
                    ]
                )
                input[0] = DEACON_INDEX.out.index.join(filter_fastq_ch)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.log[0][1]).name,
                    process.out.fastq_filtered
                ).match()},
                { assert snapshot(process.out.versions).match("versions") }
            )
        }

    }

    test("sarscov2 - fastq - single-end - stub") {

        options "-stub"

        setup {
            run("DEACON_INDEX") {
                script "../../index/main.nf"
                process {
                    """
                    input[0] = [
                        [ id:'test', single_end:true ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                    ]
                    """
                }
            }
        }

        when {
            process {
                """
                filter_fastq_ch = Channel.of(
                    [
                        [ id:'test', single_end:true ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                    ]
                )
                input[0] = DEACON_INDEX.out.index.join(filter_fastq_ch)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out,
                    process.out.versions.collect { path(it).yaml }
                ).match()}
            )
        }
    }

}
