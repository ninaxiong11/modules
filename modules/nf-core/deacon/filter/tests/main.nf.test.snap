{
    "versions": {
        "content": [
            [
                "versions.yml:md5,09ed3f492ad6f43e159b437b91d965ef"
            ]
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-29T12:20:45.59941863"
    },
    "sarscov2 - fastq - single-end - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.fq:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,09ed3f492ad6f43e159b437b91d965ef"
                ],
                "fastq_filtered": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.fq:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,09ed3f492ad6f43e159b437b91d965ef"
                ]
            },
            [
                {
                    "DEACON_FILTER": {
                        "deacon": "0.5.0"
                    }
                }
            ]
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-29T12:16:34.970620286"
    },
    "sarscov2 - fastq - paired-end": {
        "content": [
            "test.json",
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test_1.fq:md5,bf1aa22249e7e2b462c6be1a53ac55e9",
                        "test_2.fq:md5,d3106ad7395c78a212e6214177d5c054"
                    ]
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-29T12:20:45.589582528"
    },
    "sarscov2 - fastq - single-end": {
        "content": [
            "test.json",
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test.fq:md5,2250e99ec61fca0e28e2dcb5124d1a16"
                ]
            ],
            [
                {
                    "DEACON_FILTER": {
                        "deacon": "0.5.0"
                    }
                }
            ]
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-29T12:16:21.464845233"
    }
}